outputs.tf

```diff
@@ -84,24 +84,24 @@ output "oidc_provider_arn" {
 output "workers_asg_arns" {
   description = "IDs of the autoscaling groups containing workers."
   value = concat(
-    aws_autoscaling_group.workers.*.arn,
-    aws_autoscaling_group.workers_launch_template.*.arn,
+    values(aws_autoscaling_group.workers)[*].arn,
```

Comment:
I'm not sure if order will be retained here?

Reply:
From what I can gather, I think so, because each set of workers will now be sorted by name, instead of its original input order, from the variables provided by the user.

> _values are returned in lexicographical order by their corresponding keys_
> https://www.terraform.io/docs/language/functions/values.html

Here's a few of my thoughts on possible ways to solve that
- If we want to preserve that order at all costs, we could save the original order as a key in our objects, then sort using this key at the output level.
- Since were introducing a potential breaking change in the usage of this output, we could take the time to change them to maps, with their respective names as keys, or other strategy.
- If we want to minimise impact, we could leave it like this so that only the order is unpreserved, and make the necessary announcements in the release/migration process

Reply:
> If we want to preserve that order at all costs, we could save the original order as a key in our objects, then sort using this key at the output level.

I think we might need to, to make this an easy merge. 

> Since were introducing a potential breaking change in the usage of this output, we could take the time to change them to maps, with their respective names as keys, or other strategy.

Agreed, and even if we do need to keep the old array in order then we should add this as a separate output.

Reply:
I chose to preserve the ordering to make this _an easy merge_ like you said. more outputs can be added later :)